documentation/modules/ostack-token-auth.adoc

```diff
@@ -0,0 +1,95 @@
+// Module included in the following assemblies:
+//
+// * documentation/doc-Migration_Toolkit_for_Virtualization/master.adoc
+
+:_content-type: PROCEDURE
+[id="openstack-token-authentication_{context}"]
+= Using token authentication with an {osp} source provider
+
+You can use token authentication instead of username and password authentication when you use {project-first} to migrate a virtual machine that has an {osp} source provider.
+
+{project-short} supports both of the following types of token authentication:
+
+* Token with user ID
+* Token with user name
+
+For each type of token authentication, you need to use data from OpenStack to create a `Secret` manifest.
+
+.Prerequisites
+
+Have an {osp} account.
+
+.Procedure
+
+. In the dashboard of the {osp} web console, click *Project > API Access*.
+. Expand *Download OpenStack RC file* and click *OpenStack RC file*.
++
+The file that is downloaded, referred to here as `<openstack_rc_file>` includes the following fields used for token authentication:
++
+[source, terminal]
+----
+OS_AUTH_URL
+OS_PROJECT_ID
+OS_PROJECT_NAME
+OS_DOMAIN_NAME
+OS_USERNAME
+----
+
+. To get the data needed for token authentication, run the following command:
++
+[source,terminal]
+----
+$ openstack token issue
+----
++
+The output, referred to here as `<openstack_token_output>`, includes the `token`, `userID`, and `projectID` you need for authentication using a token with user ID.
+
+
+. Create a `Secret` manifest similar to the following:
+
+** For authentication using a token with user ID:
++
+[source,yaml]
+----
+cat << EOF | oc apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openstack-secret-tokenid
+  namespace: openshift-mtv
+  labels:
+    createdForProviderType: openstack
+type: Opaque
+stringData:
+  authType: token
+  token: <token_from_openstack_token_output>
+  projectID: <projectID_from_openstack_token_output>
+  userID: <userID_from_openstack_token_output>
+  url: <OS_AUTH_URL_from_openstack_rc_file>
+EOF
+----
+
+** For authentication using a token with user name:
++
+[source,yaml]
+----
+cat << EOF | oc apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openstack-secret-tokenname
+  namespace: openshift-mtv
+  labels:
+    createdForProviderType: openstack
+type: Opaque
+stringData:
+  authType: token
+  token: <token_from_openstack_token_output>
+  domainName: <OS_DOMAIN_NAME_from_openstack_rc_file>
+  projectName: <OS_PROJECT_NAME_from_openstack_rc_file>
+  username: <OS_USERNAME_from_openstack_rc_file>
+  url: <OS_AUTH_URL_from_openstack_rc_file>
+EOF
+----
+
+. Continue migrating your virtual machine according to the procedure in link:https://access.redhat.com/documentation/en-us/migration_toolkit_for_virtualization/2.4/html-single/installing_and_using_the_migration_toolkit_for_virtualization/index#migrating-virtual-machines-from-cli[Migrating virtual machines], starting with step 2, "Create a `Provider` manifest for the source provider."
```

Comment:
the link shouldn't point to 2.4

Reply:
please can we look at an `xref` instead?

Reply:
I'm adding xrefs -- Can't use xrefs in modules in OpenStack, so there is a workaround for links that adds VersionNumber as a variable in a link, instead of pointing to the last published version. We use that workaround in MTA/MTR. Not sure why it wasn't added here before. I can show you the workaround, @anarnold97, but it's less intuitive than xrefs.

Reply:
Looking at the `master.adoc`, it looks like a stripped down xref with only the id as link is used.

![image](https://github.com/kubev2v/forklift-documentation/assets/106804821/eaf5a406-a780-4f71-9580-f122efd358ff)

Please can you try:
```
xref:migrating-virtual-machines-from-cli[Migrating virtual machines]
```

Reply:
Messed up the xrefs I tried to insert and have left the link in until I talk with Andy.

Reply:
@anarnold97 Perfect!